### ISSUE TITLE
Overriding default rules

### ISSUE THREAD

**Issue by lluia** [microsoft/dtslint#297]
## Summary

Hi 👋🏻 ,

I'm using **`dstlint`** to help type an existing library, [**`next-auth`**],(https://github.com/iaincollins/next-auth) that runs on top of **NextJS**:

https://github.com/iaincollins/next-auth/pull/220

The library depends on a few types from **NextJS**...
```ts
import type { NextApiRequest, NextApiResponse } from 'next';
// ...
function NextAuth(req: NextApiRequest, res: NextApiResponse, options?: InitOptions): Promise<void>;
```
When running **`dtslint`**, it tries to parse the types from **NextJS** and throws an error given there's a `// ts-ignore` comment on them:
```
$ npx dtslint types
Error: At /Users/lluis.agusti/Code/next-auth/node_modules/next/types/index.d.ts:{"line":14,"character":6}: 'ts-ignore' is forbidden.
    at /Users/lluis.agusti/Code/next-auth/node_modules/dtslint/bin/index.js:198:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/lluis.agusti/Code/next-auth/node_modules/dtslint/bin/index.js:6:58) 
```
Following the docs, I tried to override it by [writing my own `tslint.json`](https://github.com/iaincollins/next-auth/pull/220/files#diff-65d6de72f3f0a20f873033fe5b908bb6) but it didn't make the error go away.

Any chance we could get some help on this?

**Comment by orta** [comment]
We shouldn't raise errors like this for .d.ts files inside node_modules IMO

**Comment by Haroenv** [comment]
Where would the code be for that? 

https://github.com/microsoft/dtslint/blob/285e5a9a168800e72382101b2e416baf129e8fc4/src/lint.ts#L51-L55

but then add it behind a feature flag to not lint "dependencies" at all? Or am I misunderstanding the code?

**Comment by afontcu** [comment]
Hi! Stumbled upon this issue because I'm facing the same problem – a broken CI pipeline due to a dependency usage of `ts-ignore`. Has anyone found a valid workaround? 😃  Adding a local tslint.json and disabling the check did not work.

**Comment by JounQin** [comment]
related https://github.com/microsoft/dtslint/pull/89#issuecomment-782581190
